solvebio/resource/apiresource.py

```diff
@@ -312,9 +312,21 @@ def save(self):
 
     def serialize(self, obj):
         params = {}
-        if obj._unsaved_values:
-            for k in obj._unsaved_values:
-                if k == 'id' or k == self.ID_ATTR:
-                    continue
-                params[k] = getattr(obj, k) or ""
+        unsaved_values = obj._unsaved_values
+
+        # adding 'metadata' attribute to be serialized
+        if getattr(obj, 'metadata'):
+            unsaved_values.add('metadata')
```

Comment:
Maybe this is overkill,  but there are other fields that are nested and update-able. We could add a list to each `resource` class and loop through that list. Then it is not so "hard-coded"

`metadata` is the most common one, but a quick grep in the API code also shows `Beacon.params` and `SavedQuery.params` and lots of others (though they may not be editable once set)

Reply:
I agree. That is the reason why I highlighted it was a "quick and dirty" solution. Let me think about this, but the idea with an additional list makes sense indeed.

Reply:
ya on second though, we can add in the future if it comes up. 

Would be similar class attribute to the RESOURCE_VERSION or LIST_FIELDS https://github.com/solvebio/solvebio-python/blob/master/solvebio/resource/datasetmigration.py#L22

Reply:
You are the boss... :)